code/modules/research/bepis.dm

```diff
@@ -0,0 +1,225 @@
+		say("Experiment concluded with zero product viability. Cause of error: [error_cause]")
+		banked_cash = 0
+		flick("chamber_flash",src)
+		update_icon_state()
+	return
+
+/obj/machinery/rnd/bepis/update_icon_state()
+	if(is_operational() && (powered == TRUE))
+		chamber_status = 1
+	if(((powered == FALSE) && (banked_cash == 0)) || (!is_operational()))
+		chamber_status = 0
+	if((powered == TRUE) && (banked_cash > 0) && (is_operational()))
+		chamber_status = 2
+	if (((powered == FALSE) && (banked_cash > 0)) || (banked_cash > 0) && (!is_operational()))
+		chamber_status = 3
+	if(panel_open == TRUE)
+		chamber_status = 4
+	switch(chamber_status)
+		if(0)
+			icon_state = "chamber"
+		if(1)
+			icon_state = "chamber_active"
+		if(2)
+			icon_state = "chamber_active_loaded"
+		if(3)
+			icon_state = "chamber_loaded"
+		if(4)
+			icon_state = "chamber_open"
+
+/obj/machinery/rnd/bepis/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	var/mob/living/carbon/human/H   	//the person using the console in each instance.
+	var/obj/item/card/id/Card   		//the account of the person using the console.
+	account = null
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "bepis", name, 500, 400, master_ui, state)
+		ui.open()
+	if(ishuman(user))
+		H = user
+		Card = H.get_idcard(TRUE)
+		if(Card.registered_account)
+			account = Card.registered_account
+	RefreshParts()
+
+/obj/machinery/rnd/bepis/ui_data(mob/user)
+	var/list/data = list()
+	var/silicon_check = FALSE
+	if(issilicon(user))
+		silicon_check = TRUE
+	else
+		silicon_check = FALSE
+	data["amount"] = banking_amount
+	data["stored_cash"] = banked_cash
+	data["mean_value"] = major_threshold
+	data["error_name"] = error_cause
+	data["power_saver"] = power_saver
+	data["accuracy_percentage"] = inaccuracy_percentage * 100
+	data["positive_cash_offset"] = positive_cash_offset
+	data["negative_cash_offset"] = negative_cash_offset
+	data["manual_power"] = powered ? FALSE : TRUE
+	data["silicon_check"] = silicon_check ? FALSE : TRUE
+	return data
+
+/obj/machinery/rnd/bepis/ui_act(action,params)
+	if(..())
+		return
+	switch(action)
+		if("deposit_cash")
+			if(powered == FALSE)
+				return
+			depositcash()
+			update_icon_state()
+		if("begin_experiment")
+			if(powered == FALSE)
+				return
+			if(banked_cash == 0)
+				say("Please deposit funds to begin testing.")
+				return
+			calcsuccess()
+			use_power(10000 * power_saver) //This thing should eat your APC battery if you're not careful.
```

Comment:
This is in Joule, so 10000 is a lot less than you think it is. The standard APC cell has a charge of 2500000J
Also magic number, make it a define.